File: Binary-Search/lower-bound.py
def lowerBound(A: [int], n: int, x: int) -> int:
  lo, hi = 0, n
  while lo < hi:
    mid = lo + (hi - lo) // 2
    if A[mid] < x:
      lo = mid + 1
    else:
      hi = mid
  return lo

File: Binary-Search/lower-bound.cpp
int lb(vector<int>& A, int x) {
		int lo=0,hi=A.size();
		while (lo<hi) {
				int mid=(lo+hi)>>1;
				if (A[mid]<x) lo=mid+1;
				else hi=mid;
		}
		return lo;
}